packages/web-components/src/horizontalbarchart/horizontalbarchart.ts

```diff
@@ -0,0 +1,561 @@
+  public legendContainer!: HTMLDivElement;
+  /**
+   * @public
+   * The type of the dialog modal
+   */
+  @attr
+  public _isRTL: boolean = false;
+
+  @attr({ converter: jsonConverter })
+  public data!: ChartProps[];
+
+  @attr
+  public uniqueLegends: ChartDataPoint[] = [];
+
+  @attr
+  public hideRatio = false;
+
+  private barHeight: number = 12;
+  private _selectedLegend: string = '';
+
+  constructor() {
+    super();
+  }
+
+  private bindEvents() {}
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    validateChartPropsArray(this.data, 'data');
+
+    this.initializeData();
+    this.renderChart();
+    this.renderLegends();
+  }
+
+  private initializeData() {
+    if (this.variant === Variant.SingleBar) {
+      this._hydrateData();
+    }
+    this.hydrateLegends();
+  }
+
+  public renderChart() {
+    const chartContainerDiv = d3Select(this.chartContainer);
+    chartContainerDiv
+      .selectAll('div')
+      .data(this.data!)
+      .enter()
+      .append('div')
+      .each((d, i, nodes) => {
+        this.createSingleChartBars(d, i, nodes);
+
+        //Get the tabster attributes
+        const attributes = getTabsterAttribute({ root: {} });
+
+        //Apply attributes directly to the current node
+        if (attributes[TABSTER_ATTRIBUTE_NAME] !== undefined) {
+          nodes[i].setAttribute(TABSTER_ATTRIBUTE_NAME, attributes[TABSTER_ATTRIBUTE_NAME]);
+        }
+      });
+  }
+
+  public renderLegends() {
+    const legendButtonRefs: any = [];
+    this.uniqueLegends?.forEach((d, index) => {
+      const button = document.createElement('button');
+      this.legendContainer.appendChild(button);
+      button.classList.add('legend');
+      // Store a reference to the button
+      legendButtonRefs[index] = button;
+
+      const legendRect = document.createElement('div');
+      button.appendChild(legendRect);
+      legendRect.classList.add('legendRect');
+      legendRect.style['backgroundColor'] = d.color!;
+      legendRect.style['borderColor'] = d.color!;
+
+      const legendText = document.createElement('div');
+      button.appendChild(legendText);
+      legendText.textContent = d.legend!;
+      legendText.classList.add('legendText');
+    });
+
+    const bars = this.chartContainer.querySelectorAll<HTMLElement>('.bar');
+
+    for (let i = 0; i < legendButtonRefs.length; i++) {
+      legendButtonRefs[i].addEventListener('mouseover', () => {
+        if (this._selectedLegend !== '') {
+          return;
+        }
+
+        for (let j = 0; j < bars.length; j++) {
+          if (bars[j].getAttribute('barinfo') !== this.uniqueLegends[i].legend) {
+            bars[j].style['opacity'] = '0.1';
+          } else {
+            bars[j].style['opacity'] = '1';
+          }
+        }
+        for (let j = 0; j < legendButtonRefs.length; j++) {
+          const legendRect = (
+            legendButtonRefs[j].getElementsByClassName('legendRect') as HTMLCollectionOf<HTMLDivElement>
+          )[0];
```

Comment:
If you're going to always select the first element, it might be more performant and more legible to use `querySelector` here to return the first matching Node instead of a NodeList.

```suggestion
          const legendRect = legendButtonRefs[j].querySelector('.legendRect')
```

This applies to the entire file.

Reply:
fixed in #33169